### PR TITLE
VFB-62: Edit shopping list pdf to match old layout

### PR DESCRIPTION
**Pull request by minalui**
## What's changed
- Matching formatting of shopping list with existing shopping list
- Add truck or building icon depending on delivery or collection on shopping list and shipping labels (absorbed [VFB-56](https://softwiretech.atlassian.net/jira/software/c/projects/VFB/boards/862?selectedIssue=VFB-56))
- Fix bug for notes and quantity not showing in pdf

## Screenshots / Videos
| Before     | After      |
|------------|------------|
| image image | image image |

## Checklist
- [x] The ticket is up-to-date - Please document any deviations from the original approach if there is any.
- [x] I have documented the testing steps for QA
- [x] I have self-reviewed this PR
- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've verified it works via `npm run build` and `npm run start`
- [x] Make sure you've fixed all linting problems with `npm run lint_fix`
- [x] Make sure you've tested via `npm run test:e2e`

If you have made any changes to the database... N/A
  - [ ] The migration files are up-to-date with my final set up (`npx supabase db diff -f <name_of_migration>` should create nothing at this point)
  - [ ] I have updated the typescript definitions for the database with `db:local:generate_types`
  - [ ] I have modified the seed in `supabase/seed/seed.mts` if appropriate
  - [ ] If I have modified the seed, I have also generated the seed with `npm run db:generate_seed` 
  - [ ] With my final set up, I can run `npm run dev:reset_supabase` without any errors.
  - Does this require resetting the deployed database? - YES / NO (remove as appropriate)
  - [ ] I have checked that migration can happen successfully without resetting the database, doing the following.
    - Make sure you have rebased your branch onto dev or merged dev into your branch.
    - Checkout the dev branch
    - Run `npm run post_checkout` to reset the database, including the seed data.
    - Checkout your branch
    - Run `npx supabase migrations up --include-all --local` to run all outstanding migration files
    - Check that the resulting database is what you expect, bar any seed data changes.

### CODE REVIEW

File: src/pdf/ShoppingList/shoppingListPdfDataProps.ts [hunk 1]
```diff
@@ -40,8 +40,8 @@ const getQuantityAndNotes = (
     if (size >= 10) {
         size = 10;
     }
-    const sizeQuantity = `${size}_quantity` as keyof Schema["lists"];
-    const sizeNotes = `${size}_notes` as keyof Schema["lists"];
+    const sizeQuantity = `quantity_for_${size}` as keyof Schema["lists"];
+    const sizeNotes = `notes_for_${size}` as keyof Schema["lists"];
```
Comment (rickaw) [inline src/pdf/ShoppingList/shoppingListPdfDataProps.ts:44]: I think these `as keyof Schema["lists"]` is what allowed this bug to happen in the first place, so let's try make this more robust.

One approach would be to define the set of allowed family sizes, and make sure the family size is one of them, before computing the quantity and notes strings. This way, we don't need to use `as` and this kind of bug won't happen again.

```ts
const allowedFamilySizes = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10] as const;
type FamilySize = (typeof allowedFamilySizes)[number];

function numberIsValidFamilySize(value: number): value is FamilySize {
    for (const allowedFamilySize of allowedFamilySizes) {
        if (allowedFamilySize === value) {
            return true;
        }
    }
    return false;
}

const getQuantityAndNotes = (
    row: Schema["lists"],
    size: number
): Pick<Item, "quantity" | "notes"> => {
    if (size >= 10) {
        size = 10;
    }

    if (!numberIsValidFamilySize(size)) {
        // return an error as we normally do
    }

    const sizeQuantity: keyof Schema["lists"] = `quantity_for_${size}`;
    const sizeNotes: keyof Schema["lists"] = `notes_for_${size}`;
    return {
        quantity: row[sizeQuantity]?.toString() ?? "",
        notes: row[sizeNotes]?.toString() ?? "",
    };
};
```

File: src/pdf/ShippingLabels/ShippingLabelsPdf.tsx [hunk 2]
```diff
@@ -118,13 +120,20 @@ const LabelCard: React.FC<LabelCardProps> = ({ data, index, quantity }) => {
...
-                                ? data.collection_centre // TODO VFB-56 needs icon
-                                : "Delivery"}
+                                ? data.collection_centre + " "
+                                : "Delivery "}
                         </Text>
+                        <FontAwesomeIconPdfComponent
+                            faIcon={
+                                data.collection_centre !== "DLVR"
+                                    ? faBuildingCircleArrowRight
+                                    : faTruck
```
Comment (rickaw) [inline src/pdf/ShippingLabels/ShippingLabelsPdf.tsx:134]: the logic is probs clearer this way?
```suggestion
                                data.collection_centre === "DLVR"
                                    ? faTruck
                                    : faBuildingCircleArrowRight
```

File: src/pdf/FontAwesomeIconPdfComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+                    <Path
+                        d={drawnPath}
+                        key="Font Awesome Icon"
+                        fill={color}
+                        fillOpacity={duotone && indexNumber === 0 ? 0.4 : 1.0}
+                    />
+                ))}
+        </Svg>
+    );
+};
```
[the remark this answers is not in the document]
Comment (minalui) [reply src/pdf/FontAwesomeIconPdfComponent.tsx:26]: image
Each font awesome icon is an Icon Definition - so I think icon[4] is string eg. unicode.
I don't know svg's very well but I got the solution from here: https://github.com/diegomura/react-pdf/discussions/1334

[the remark this answers is not in the document]
Comment (minalui) [reply src/pdf/FontAwesomeIconPdfComponent.tsx:26]: Or we can use an image instead like the logo?

[the remark this answers is not in the document]
Comment (rickaw) [reply src/pdf/FontAwesomeIconPdfComponent.tsx:26]: Can we make this clearer in the code? e.g. 
```ts
const width = icon[0]
const height = icon[1]
const isDuoTone = Array.isArray(icon[4])
```
etc

**Review by rickaw — APPROVED** [review]

**Pull request merged**